gprofiler/profilers/registry.py

```diff
@@ -80,26 +95,88 @@ def register_profiler(
         disablement_help = f"Disable the runtime-profiling of {profiler_name} processes"
 
     def profiler_decorator(profiler_class: Any) -> Any:
-        assert profiler_name not in profilers_config, f"{profiler_name} is already registered!"
+        assert profiler_name is not None, "Profiler name must be defined"
+        assert profiler_name not in (
+            config.profiler_name for profilers in profilers_config.values() for config in profilers
+        ), f"{profiler_name} is already registered!"
         assert all(
-            arg.dest.startswith(profiler_name.lower()) for arg in profiler_arguments or []
-        ), f"{profiler_name}: Profiler args dest must be prefixed with the profiler name"
-        profilers_config[profiler_name] = ProfilerConfig(
-            profiler_mode_argument_help,
-            disablement_help,
-            profiler_class,
-            possible_modes,
-            supported_archs,
-            supported_profiling_modes,
-            supported_windows_archs,
-            default_mode,
-            profiler_arguments,
-        )
+            arg.dest.startswith(runtime.lower()) for arg in profiler_arguments or []
+        ), f"{profiler_name}: Profiler args dest must be prefixed with the profiler runtime name"
+        profilers_config[runtime] += [
+            ProfilerConfig(
+                profiler_name,
+                runtime,
+                is_preferred,
+                profiler_mode_argument_help,
+                disablement_help,
+                profiler_class,
+                possible_modes,
+                supported_archs,
+                supported_profiling_modes,
+                supported_windows_archs,
+                default_mode,
+                profiler_arguments,
+            )
+        ]
         profiler_class.name = profiler_name
         return profiler_class
 
     return profiler_decorator
 
 
-def get_profilers_registry() -> Dict[str, ProfilerConfig]:
+def get_profilers_registry() -> Dict[str, List[ProfilerConfig]]:
     return profilers_config
+
+
+def get_profilers_by_name() -> Dict[str, ProfilerConfig]:
+    return {config.profiler_name: config for configs in profilers_config.values() for config in configs}
+
+
+def get_runtime_possible_modes(runtime: str) -> List[str]:
+    possible_modes: List[str] = [ProfilerConfig.ENABLED_MODE] if len(profilers_config[runtime]) > 1 else []
+    for config in profilers_config[runtime]:
+        possible_modes += [m for m in config.get_active_modes() if m not in possible_modes]
```

Comment:
Instead of checking for existence you can build it as a `set` and then convert to a list.